### DISCUSSION TITLE
good example for how to use portsChanged

### DISCUSSION THREAD

**Discussion by cristianvogel** [djipco/webmidi#215] [Q&A]
Hi!

Would it be too much to ask for a code example of how to listen to portsChanged? 

https://webmidijs.org/api/classes/WebMidi#event-portschanged

I can't really figure out how to set up a listener for this event, because it is not coming from a midi port? I specifically want to know if the output ports have changed.

**Comment by djipco** [comment]
Hello Christian,

When listening to an event, you have to add the listener on the object that dispatches the event. In this case, the event is being dispatched by the `WebMidi` object, so this is how you would do it:

```javascript
WebMidi.addListener("portschanged", e => {
   console.log(e)
});
```
Hope this helps!
